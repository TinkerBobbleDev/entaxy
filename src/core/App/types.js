export default {
  AUTHENTICATION_LOGIN: 'AUTHENTICATION_LOGIN'
}
