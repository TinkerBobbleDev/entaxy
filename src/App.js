import React from 'react';
import Header from './common/Header';
import Routes from './routes';
import 'typeface-roboto'

const App = ({ history }) => (
  <div>
    <Header/>
    <Routes history= {history}/>
  </div>
)

export default App
