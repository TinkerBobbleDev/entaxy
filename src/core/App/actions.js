import types from './types'

const authenticationLogin = profile => ({
  type: types.AUTHENTICATION_LOGIN,
  profile
})

export { authenticationLogin }

export default {
  authenticationLogin
}
