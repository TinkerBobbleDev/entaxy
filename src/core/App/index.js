import React from 'react'
// import PropTypes from 'prop-types';
import Header from '../../common/Header'
import Routes from '../../routes'
import 'typeface-roboto'
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles'
import Reboot from 'material-ui/Reboot'
import createHistory from 'history/createBrowserHistory'
import blueGrey from 'material-ui/colors/blueGrey';
import teal from 'material-ui/colors/teal';

const history = createHistory()


const theme = createMuiTheme({
  palette: {
    primary: { main: blueGrey[800] },
    secondary: { main: teal[500] },
  },
});

const App = (props) => {
  // const { classes } = props
  return (
    <Reboot>
      <MuiThemeProvider theme={theme}>
        <Header/>
		    <Routes history= {history}/>
      </MuiThemeProvider>
    </Reboot>
  )
}

// App.propTypes = {
//   classes: PropTypes.object.isRequired,
// };

export default App
