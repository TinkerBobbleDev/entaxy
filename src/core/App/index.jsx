import React from 'react'
import { createMuiTheme, MuiThemeProvider } from 'material-ui/styles'
import Reboot from 'material-ui/Reboot'
import blueGrey from 'material-ui/colors/blueGrey'
import teal from 'material-ui/colors/teal'
import Header from '../../common/Header/index'
import Routes from '../../routes'
import 'typeface-roboto'

// import PropTypes from 'prop-types';
const theme = createMuiTheme({
  palette: {
    primary: { main: blueGrey[800] },
    secondary: { main: teal[500] }
  }
})

const App = () => {
  // const { classes } = props
  return (
    <Reboot>
      <MuiThemeProvider theme={theme}>
        <Header />
        <Routes />
      </MuiThemeProvider>
    </Reboot>
  )
}

// App.propTypes = {
//   classes: PropTypes.object.isRequired,
// };

export default App
