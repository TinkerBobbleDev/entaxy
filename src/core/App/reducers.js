import types from './types'

const app = (state = {}, action) => {
  switch (action.type) {
    case types.AUTHENTICATION_LOGIN:
      return {
        ...state,
        loggedIn: true,
        profile: action.profile
      }

    default:
      return state
  }
}

export default app
