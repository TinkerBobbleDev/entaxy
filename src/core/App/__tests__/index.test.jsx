import React from 'react'
import { shallow } from 'enzyme'
import App from '../'

jest.mock('../../../common/InstitutionIcon', () => 'InstitutionIcon')

describe('App', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.debug()).toMatchSnapshot()
  })
})
