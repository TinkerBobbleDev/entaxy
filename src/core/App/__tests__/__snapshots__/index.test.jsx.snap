// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-8"
  >
    <div
      className="MuiToolbar-root-37 MuiToolbar-gutters-38"
    >
      <img
        alt="Entaxy logo"
        className="Logo-logo-39"
        src="logo.png"
      />
      <h2
        className="MuiTypography-root-40 MuiTypography-title-46 MuiTypography-colorInherit-59 Header-flex-1"
      >
        Entaxy
      </h2>
      <button
        className="MuiButtonBase-root-80 MuiButton-root-65 MuiButton-colorInherit-69 login"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-66"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-83"
        />
      </button>
    </div>
  </header>,
  <div
    className="Taxes-root-90"
  >
    <div
      className="MuiGrid-typeContainer-93"
    >
      <div
        className="MuiGrid-typeItem-94 MuiGrid-grid-xs-10-128"
      >
        <div
          className="MuiPaper-root-10 MuiPaper-elevation2-14 MuiPaper-rounded-11 Taxes-paper-91 Taxes-chart-92"
        >
          <h1
            className="MuiTypography-root-40 MuiTypography-headline-45 MuiTypography-gutterBottom-57 MuiTypography-alignCenter-53"
          >
            Taxes for 
            Ontario
             
            2017
          </h1>
          <div
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <div
        className="MuiGrid-typeItem-94 MuiGrid-grid-xs-2-120"
      >
        <div
          className="MuiPaper-root-10 MuiPaper-elevation2-14 MuiPaper-rounded-11 Taxes-paper-91"
        >
          <div
            className="MuiFormControl-root-183 MuiFormControl-fullWidth-186"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="MuiFormLabel-root-192 MuiInputLabel-root-187 MuiInputLabel-formControl-188 MuiInputLabel-animated-191 MuiInputLabel-shrink-190"
              data-shrink={true}
              htmlFor="income"
            >
              Your income
            </label>
            <div
              className="MuiInput-root-198 MuiInput-formControl-199 MuiInput-underline-202"
            >
              <input
                aria-invalid={false}
                aria-label="Your income for 2017"
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-206"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={undefined}
                onMouseUp={[Function]}
                placeholder={undefined}
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="text"
                value="$90,000"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root-10 MuiPaper-elevation2-14 MuiPaper-rounded-11 Taxes-paper-91"
        >
          <h3
            className="MuiTypography-root-40 MuiTypography-subheading-47"
          >
            Tax credits
          </h3>
          <div
            className="MuiFormControl-root-183 MuiFormControl-fullWidth-186"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <label
              className="MuiFormLabel-root-192 MuiInputLabel-root-187 MuiInputLabel-formControl-188 MuiInputLabel-animated-191 MuiInputLabel-shrink-190"
              data-shrink={true}
              htmlFor="rrsp"
            >
              RRSP contribution
            </label>
            <div
              className="MuiInput-root-198 MuiInput-formControl-199 MuiInput-underline-202"
            >
              <input
                aria-invalid={false}
                aria-label="RRSP for 2017"
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-206"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={undefined}
                onMouseUp={[Function]}
                placeholder={undefined}
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="text"
                value="$10,000"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiPaper-root-10 MuiPaper-elevation2-14 MuiPaper-rounded-11 Taxes-paper-91"
        >
          <p
            className="MuiTypography-root-40 MuiTypography-body1-49"
          >
            <strong>
              Taxable income:
            </strong>
             
            CA$80,000.00
          </p>
          <p
            className="MuiTypography-root-40 MuiTypography-body1-49"
          >
            <strong>
              Tax amount:
            </strong>
             
            CA$17,205.49
          </p>
          <p
            className="MuiTypography-root-40 MuiTypography-body1-49"
          >
            <strong>
              Marginal Tax Rate:
            </strong>
             
            29.65%
          </p>
          <p
            className="MuiTypography-root-40 MuiTypography-body1-49"
          >
            <strong>
              Average Tax Rate:
            </strong>
             
            21.51%
          </p>
        </div>
      </div>
    </div>
  </div>,
]
`;
